batch/indexer_TransferApproval.py

```diff
@@ -165,24 +188,39 @@ def get_block_timestamp(event) -> datetime:
         block_timestamp = web3.eth.getBlock(event["blockNumber"])["timestamp"]
         return block_timestamp
 
-    def get_token_list(self):
+    def get_contract_list(self):
         self.token_list = []
+        self.exchange_list = []
         ListContract = Contract.get_contract(
             contract_name="TokenList",
             address=TOKEN_LIST_CONTRACT_ADDRESS
         )
         listed_tokens = self.db.query(Listing).all()
+
+        _exchange_list_tmp = []
         for listed_token in listed_tokens:
             token_info = ListContract.functions.getTokenByAddress(listed_token.token_address).call()
-            if token_info[1] == "IbetShare":
+            token_type = token_info[1]
+            if token_type == "IbetShare" or token_type == "IbetStraightBond":
                 token_contract = Contract.get_contract(
-                    contract_name="IbetShare",
+                    contract_name="IbetSecurityTokenInterface",
                     address=listed_token.token_address
                 )
                 self.token_list.append(token_contract)
+                tradable_exchange = token_contract.functions.tradableExchange().call()
+                _exchange_list_tmp.append(tradable_exchange)
```

Comment:
If ZERO_ADDRESS is set, it is better to SKIP.

Reply:
thnx